### DISCUSSION TITLE
Deta CRON not working

### DISCUSSION THREAD

**Discussion by robsannaa** [deta/cloud-docs#597] [Help] [answered]
hi all,I am trying to run a deta CRON job following exactly what is proposed into the docs: 

```
from deta import app

# define a function to run on a schedule
# the function must take an event as an argument
@app.lib.cron()
def cron_job(event):
    return "running on a schedule"

```

However, when I try to run the deployed function, I get the following error:

```
Response:
{
        "error": "No function found for action 'an empty action'"
}
```
I have also tried to instantiate the app like this:

```
from deta import app
app = app()

@app.lib.cron()
def cron_job(event):
    return "Hello Deta, I am a cron job"

```

But I am getting the same error. I have tried many different combinations, but all of the does not work.
What am I doing wrong?

**Comment by robsannaa** [comment]
Update: I managed to re-create a new deta app from scratch and I was able to run the example code, however, when I replace the code with my function, I get the following error: `TypeError: 'BareApp' object is not callable`

**Comment by sudo-subg** [comment, marked as answer]
I guess you are using `deta run`, don't use that. To schedule your cron use [this](https://docs.deta.sh/docs/micros/cron#set-cron) and to know more about deta crons check [here](https://docs.deta.sh/docs/micros/cron).

**Reply by robsannaa** [reply]
That was actually my mistake... thanks a lot. Is there a timeout for functions?

**Reply by sudo-subg** [reply]
Yes micros have a timeout of 10 seconds.

**Reply by robsannaa** [reply]
ohhhhh that explains a lot!
